### PR TITLE
PIE-2016: Added PoaQueryService method that returns local node signer…

### PR DESCRIPTION
**Pull request by mark-terry**
… address.

Signed-off-by: Mark Terry <mark.terry@consensys.net>

## PR description
Added PoaQueryService method that returns local node signer address.

## Fixed Issue(s)

#133

### CODE REVIEW

File: consensus/ibft/src/test/java/org/hyperledger/besu/consensus/ibft/queries/IbftQueryServiceImplTest.java [hunk 1]
```diff
@@ -48,6 +48,7 @@
 public class IbftQueryServiceImplTest {
 
   private Blockchain blockchain = mock(Blockchain.class);
```
Comment (CjHare) [inline consensus/ibft/src/test/java/org/hyperledger/besu/consensus/ibft/queries/IbftQueryServiceImplTest.java]: You could switch these two mocks over to the annotation and add the mockito runner to this test (as that is pretty consistent pattern being used throughout the tests)

File: besu/src/main/java/org/hyperledger/besu/controller/CliqueQueryPluginServiceFactory.java [hunk 2]
```diff
@@ -17,24 +17,27 @@
...
 import org.hyperledger.besu.ethereum.chain.Blockchain;
 import org.hyperledger.besu.plugin.services.metrics.PoAMetricsService;
 import org.hyperledger.besu.plugin.services.query.PoaQueryService;
 import org.hyperledger.besu.services.BesuPluginContextImpl;
 
 public class CliqueQueryPluginServiceFactory implements PluginServiceFactory {
 
-  final Blockchain blockchain;
+  private final Blockchain blockchain;
+  private final KeyPair keyPair;
```
Comment (usmansaleem) [inline besu/src/main/java/org/hyperledger/besu/controller/CliqueQueryPluginServiceFactory.java]: should `keyPair` be called `nodeKeys`?

File: plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/PoaQueryService.java [hunk 3]
```diff
@@ -36,4 +36,11 @@
    * @return The identity of the proposer for the given block.
    */
   Address getProposerOfBlock(final BlockHeader header);
+
+  /**
+   * Retrieves the signer {@link Address} of the local node.
```
Comment (CjHare) [inline plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/PoaQueryService.java]: You are not really adding any detail beyond that already provided by the method signature.
Perhaps these questions might be worth covering in the Javadoc:
- What is this local signer address in the context of the node?
- What is this local signer address in the context of the network?
- What is the intended use / would it be used?

File: besu/src/main/java/org/hyperledger/besu/controller/CliqueQueryPluginServiceFactory.java (hunk 2 above)
Comment (usmansaleem) [reply besu/src/main/java/org/hyperledger/besu/controller/CliqueQueryPluginServiceFactory.java]: or `localNodeKeyPair` as you did later.

File: consensus/common/src/main/java/org/hyperledger/besu/consensus/common/PoaQueryServiceImpl.java [hunk 4]
```diff
@@ -27,10 +28,15 @@
 
   private final BlockInterface blockInterface;
   private final Blockchain blockchain;
+  private final KeyPair keyPair;
```
Comment (usmansaleem) [inline consensus/common/src/main/java/org/hyperledger/besu/consensus/common/PoaQueryServiceImpl.java]: does it make sense to introduce `getLocalNodeKeyPair` method in either poaQueryService or poaMetricsService?

Comment (usmansaleem) [reply consensus/common/src/main/java/org/hyperledger/besu/consensus/common/PoaQueryServiceImpl.java]: , you have `getLocalSignerAddress` later on. You might get away with creating a default method in interface that calls getLocalNodeKeyPair method.

Comment (usmansaleem) [reply consensus/common/src/main/java/org/hyperledger/besu/consensus/common/PoaQueryServiceImpl.java]: not a show stopper though. Things looks OK as it is as well.

File: plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/PoaQueryService.java (hunk 3 above)
Comment (usmansaleem) [reply plugin-api/src/main/java/org/hyperledger/besu/plugin/services/query/PoaQueryService.java]: 👍

File: consensus/ibft/src/test/java/org/hyperledger/besu/consensus/ibft/queries/IbftQueryServiceImplTest.java (hunk 1 above)
Comment (mark-terry) [reply consensus/ibft/src/test/java/org/hyperledger/besu/consensus/ibft/queries/IbftQueryServiceImplTest.java]: I've seen a mix of this, but will switch these over if that's the direction we're going.  👍

File: consensus/common/src/main/java/org/hyperledger/besu/consensus/common/PoaQueryServiceImpl.java (hunk 4 above)
Comment (mark-terry) [reply consensus/common/src/main/java/org/hyperledger/besu/consensus/common/PoaQueryServiceImpl.java]: I've updated the field name.

File: consensus/common/src/main/java/org/hyperledger/besu/consensus/common/PoaQueryServiceImpl.java [hunk 5]
```diff
@@ -46,4 +52,9 @@ public Address getProposerOfBlock(final BlockHeader header) {
   protected Blockchain getBlockchain() {
     return blockchain;
   }
+
+  @Override
+  public Address getLocalSignerAddress() {
+    return org.hyperledger.besu.ethereum.core.Address.extract(localNodeKeypair.getPublicKey());
```
Comment (usmansaleem) [inline consensus/common/src/main/java/org/hyperledger/besu/consensus/common/PoaQueryServiceImpl.java]: you may get away by static importing `extract` method. Not a show stopper.

Comment (mark-terry) [reply consensus/common/src/main/java/org/hyperledger/besu/consensus/common/PoaQueryServiceImpl.java]: We've got two `Address` classes in scope - one for Besu core, one for plugin API.  I've left this one as an explicit reference to aid with context.

**Pull request merged**